shopinvader_api_payment_provider_custom/routers/payment_custom.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2024 ACSONE SA (https://acsone.eu).
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+from typing import Annotated, Any
+from urllib.parse import urljoin
+
+from fastapi import Depends, HTTPException
+
+from odoo import api, models
+
+from odoo.addons.fastapi.dependencies import odoo_env
+from odoo.addons.payment.models.payment_transaction import PaymentTransaction
+from odoo.addons.shopinvader_api_payment.routers import payment_router
+from odoo.addons.shopinvader_api_payment.routers.utils import Payable
+from odoo.addons.shopinvader_api_payment.schemas import TransactionProcessingValues
+
+_logger = logging.getLogger(__name__)
+
+
+@payment_router.post("/payment/providers/custom/pending")
+def custom_payment_pending_msg(
+    payable: str,
+    odoo_env: Annotated[api.Environment, Depends(odoo_env)],
+):
+    try:
+        payable_obj = Payable.decode(odoo_env, payable)
+    except Exception as e:
+        _logger.info("Could not decode payable")
+        raise HTTPException(403) from e
+
+    odoo_rec = odoo_env[payable_obj.payable_model].sudo().browse(payable_obj.payable_id)
+    tx_sudo = odoo_rec.transaction_ids[0]  # TODO: how to choose which one?
```

Comment:
This method doesn't seem to be the best one to find the right transaction to set to pending. Must think of a better way to do that. Take an encoded tx reference?